eth2near/eth2near-block-relay-rs/src/eth2near_relay.rs

```diff
@@ -179,21 +179,26 @@ impl Eth2NearRelay {
     }
 
     fn get_last_slot(&mut self) -> Result<u64, Box<dyn Error>> {
-        trace!(target: "relay", "= Search for last slot on near =");
+        debug!(target: "relay", "= Search for last slot on near =");
 
         let mut slot = self.eth_client_contract.get_last_submitted_slot();
 
         let finalized_slot = self.eth_client_contract.get_finalized_beacon_block_slot()?;
-        trace!(target: "relay", "Finalized slot on near={}", finalized_slot);
+        debug!(target: "relay", "Finalized slot on near={}", finalized_slot);
 
         slot = max(finalized_slot, slot);
-        trace!(target: "relay", "Init slot for search as {}", slot);
+        debug!(target: "relay", "Init slot for search as {}", slot);
 
-        if self.block_known_on_near(slot)? {
-            self.search_slot_forward(slot)
+        if slot == finalized_slot || self.block_known_on_near(slot)? {
+            while (self.block_known_on_near(slot + 1)?) {
+                slot += 1;
+            }
         } else {
-            self.search_slot_backward(finalized_slot, slot)
+            while slot > finalized_slot && !(self.block_known_on_near(slot)?) {
+                slot -= 1;
+            }
         }
+        Ok(slot)
```

Comment:
Binary search should be still available as a feature/config

Reply:
fixed: https://github.com/aurora-is-near/rainbow-bridge/pull/775/commits/2317e8f6fe7ceb681c021ea42c9a196513cd7930